modules/popup-stack/README.md

```diff
@@ -0,0 +1,181 @@
+# Canvas Kit Popup Stack
+
+Stack for managing popup UIs to coordinate global concerns like escape key handling and rendering
+order. The popup stack is a FILO stack that is framework agnostic and is shared for the entire page.
+This sharing is enforced. If the window stack variable (`window.workday.__popupStack`) has already
+been declared, the first-instantiated instance will be used. There is no need to worry about
+bootstrapping at a specific time.
+
+## What is Stacked UI?
+
+A Stacked UI (typically classified as "popups") component generally isn't statically positioned (not
+in the normal flow of the page) and sits on top of other content - usually occluding other content.
+A stacked UI element typically redirects a users attention temporarily. For example, a Modal appears
+on top of all other content while it is active. A tooltip appears on top of other content when the
+target is focused or hovered.
+
+Stacked components include:
+
+- Tooltips
+- Modal dialogs
+- Select menu dropdown
+- Toast messages
+- Popups
+- Snackbars
+- Dropdown menus
+- Windows
+
+Stacked UI components are usually unbounded by the physical boundaries of a target element through a
+technique of "portalling" the Popup to the bottommost element of the `body` element. This allows a
+Tooltip to appear outside any overflowed content. This technique does not work outside the confines
+of an iframe, however.
+
+## Why do I need a stack manager?
+
+Simple UIs may define z-index values globally per component type, so why do we need to manage
+popups? Popups usually have accessibility requirements that the popup should close if the escape key
+is pressed. The naive way of doing this is listening for the escape key in all components. This
+works if only one Popup is on the page at a time, but often UIs get more complicated. A simple
+example is a Modal dialog with a Select component inside. The user opens the Modal dialog and then
+opens a Select menu inside the Modal. The user may hit the escape key which should close the Select
+dropdown menu, but not the Modal dialog. A more painful example is the same scenario, but with click
+outside detection. The user wants to select an item within the Select dropdown menu, but the Modal
+dialog detects clicking inside the dropdown menu as an outside click because the dropdown menu isn't
+a direct child component of the Modal dialog because of the "portalling" technique commonly used.
+
+A stack manager keeps track of all Stacked UI registered on the page as they are opened and closed.
+A special Window component may be persistent and should be brought to the top of the stack when
+clicked on. This is how operating system "windows" work. The behavior requires global coordination
+to determine the order that these popups should be rendered.
+
+## Installation
+
+```sh
+yarn add @workday/canvas-kit-popup-stack
+```
+
+## Usage
```

Comment:
I think we should probably call out to the hooks here for React users

Reply:
Hooks are in the Popup module. There isn't really a good way to link to other docs currently.

Reply:
Yeah I didn't mean document them here. Can't we just do something like:

```
If you're using the Popup Stack within a React app, 
we provide [convenient hooks](../popup/react/README.md#hooks) 
for these behaviors in `@workday/canvas-kit-react-popup`
```